Mage/src/main/java/mage/abilities/keyword/WardAbility.java

```diff
@@ -75,10 +100,17 @@ public WardAbility copy() {
     @Override
     public String getRule() {
         StringBuilder sb = new StringBuilder("ward");
-        if (cost instanceof ManaCost) {
-            sb.append(' ').append(cost.getText());
+        if (cost != null) {
+            if (cost instanceof ManaCost) {
+                sb.append(' ').append(cost.getText());
+            } else {
+                sb.append("&mdash;").append(CardUtil.getTextWithFirstCharUpperCase(cost.getText())).append('.');
+            }
         } else {
-            sb.append("&mdash;").append(CardUtil.getTextWithFirstCharUpperCase(cost.getText())).append('.');
+            sb.append(" {X}");
+            if (whereXIs != null) {
+                sb.append(" where X is ").append(whereXIs);
```

Comment:
All appears to work correctly, just a minor issue with line 112, being that the appended string should be ", where X is " with the comma at the beginning, in order to generate correct rules text

Reply:
Done